File: src/main/java/capstone/bcs/lifo/controllers/SpecialOffer.java
package capstone.bcs.lifo.controllers;


import capstone.bcs.lifo.commands.LoginForm;
import capstone.bcs.lifo.util.ValidSessionDataUtil;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;

import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpSession;

@Controller
public class SpecialOffer {

    @RequestMapping("/special_offer")
    public String getPage(Model model, HttpSession session){
        model.addAttribute("LoginForm", new LoginForm());
        ValidSessionDataUtil validSDU = new ValidSessionDataUtil(session);
        model.addAttribute("cartsize",validSDU.getProductListSize());
        model.addAttribute("carttotal",validSDU.getCartTotal());
        return "special_offer";
    }

    @RequestMapping("products/special_offer")
    public String getPageHotFix(Model model, HttpSession session){
        model.addAttribute("LoginForm", new LoginForm());
        ValidSessionDataUtil validSDU = new ValidSessionDataUtil(session);
        model.addAttribute("cartsize",validSDU.getProductListSize());
        model.addAttribute("carttotal",validSDU.getCartTotal());
        return "special_offer";
    }

//    @RequestMapping("/special_offer/{id}")
//    public String getPageVar(HttpServletRequest request, @PathVariable("id") Integer id, Model model) {
//        model.addAttribute("LoginForm", new LoginForm());
//        String referer = request.getHeader("Referer"); // if change needed
//        return "redirect:" + "special_offer";
//    }


}

File: src/main/java/capstone/bcs/lifo/controllers/SpecialOffer.kt
package capstone.bcs.lifo.controllers


import capstone.bcs.lifo.commands.LoginForm
import capstone.bcs.lifo.services.ProductService
import capstone.bcs.lifo.util.SessionTransitionUtil
import capstone.bcs.lifo.util.ValidSessionDataUtil
import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import javax.servlet.http.HttpSession

// Kotlin controller

@Controller
class SpecialOfferKt(val productService: ProductService) {


    @RequestMapping("/special_offer_backup")
    fun getSpecialPage(model: Model, httpSession: HttpSession):String {
        model.addAttribute("LoginForm", LoginForm())

        // == produces the cart ==
        var sU = SessionTransitionUtil()
        var httpSession = sU.AnonSession(httpSession)
        // == produces the cart ==

        model.addAttribute("LoginForm", LoginForm())
        val validSDU = ValidSessionDataUtil(httpSession)
        model.addAttribute("cartsize", validSDU.productListSize)
        model.addAttribute("carttotal", validSDU.cartTotal)
        return "special_offer_backup"
    }

    @RequestMapping("products/special_offer/{id}")
    fun getSpecialPageHotFix(model: Model, httpSession: HttpSession,@PathVariable id:String):String {
        model.addAttribute("LoginForm", LoginForm())

        val productCat = Integer.valueOf(id)
        model.addAttribute("products", productService.getProductsByCategory(Integer.valueOf(productCat)))

        // == produces the cart ==
        var sU = SessionTransitionUtil()
        var httpSession = sU.AnonSession(httpSession)
        // == produces the cart ==

        model.addAttribute("LoginForm", LoginForm())
        val validSDU = ValidSessionDataUtil(httpSession)
        model.addAttribute("cartsize", validSDU.productListSize)
        model.addAttribute("carttotal", validSDU.cartTotal)
        return "special_offer"
    }



}